### ISSUE TITLE
Compile cache files show up as inside app.asar

### ISSUE THREAD

**Issue by Arcanemagus** [atom/atom#14010]
### Prerequisites
_Note: Impossible to test under safe mode as by definition this requires external packages._

* [ ] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: http://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode
    * Followed all applicable steps in the debugging guide: http://flight-manual.atom.io/hacking-atom/sections/debugging/
    * Checked the FAQs on the message board for common solutions: https://discuss.atom.io/c/faq
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom
    * Checked that there is not already an Atom package that provides the described functionality: https://atom.io/packages

### Description

Files from packages that are compiled are showing up as being inside the `app.asar` when searching for them in the Developer Tools. The real locations only open the compiled version.

### Steps to Reproduce

1. Have a third-party package installed.
2. Open Developer Tools
3. Attempt to open a file from the search (since virtually no files show up in the Sources list...)

_Note: I was seeing even stranger behavior than this, with breakpoints on package files setting themselves on internal Atom files. I deleted the `compile-cache` folder which leads me to the current better, but still very broken behavior._

**Expected behavior:** [What you expect to happen]

The file you select from the search opens the original source file, with the source map making breakpoints work.

**Actual behavior:** [What actually happens]

File "in" `app.asar` is the real file, with the source map showing which parts of the file aren't in the in-use code.
File at the real location is the compiled version of the file, when you open this it tells you there is a source map active.

Note that it is only possible to get to the real version of a package file if it has a unique name. If you try to search for a file name used in many packages (like `main.js`) then it seems to pick whichever shows up first in the list of real results.

**Reproduces how often:** [What percentage of the time does it reproduce?]

100% of the time.

### Versions

You can get this information from copy and pasting the output of `atom --version` and `apm --version` from the command line. Also, please include the OS and what version of the OS you're running.

```
> atom --version
Atom    : 1.16.0-beta0
Electron: 1.3.13
Chrome  : 52.0.2743.82
Node    : 6.5.0
> apm --version
apm  1.16.1
npm  3.10.5
node 6.9.5 x64
python 2.7.12
git 2.12.0.windows.1
visual studio 2015
```

**Comment by Arcanemagus** [comment]
This appears to be fixed on a build based off of merging 842b3f0f3cb39562914f7f5e6bd071acad9c1e3b into 25dcf2ed81e920fbdeb0f1cf7684269e75f41289.

**Comment by Arcanemagus** [comment]
Confirming this as fixed in v1.17.0-beta3.
